src/components/DayPicker.jsx

```diff
@@ -395,24 +329,23 @@ export default class DayPicker extends React.Component {
   }
 
   onPrevMonthClick(nextFocusedDate, e) {
-    const { isRTL } = this.props;
+    const { numberOfMonths, isRTL } = this.props;
+    const { calendarMonthWidth } = this.state;
 
     if (e) e.preventDefault();
 
     let translationValue =
-      this.isVertical() ? this.getMonthHeightByIndex(0) : this.dayPickerWidth;
+      this.isVertical() ? this.calendarMonthHeights[0] : calendarMonthWidth;
 
-    if (isRTL && this.isHorizontal()) {
-      translationValue = -2 * this.dayPickerWidth;
-    }
+    if (this.isHorizontal()) {
+      if (isRTL) {
+        translationValue = -2 * calendarMonthWidth;
+      }
 
-    // The first CalendarMonth is always positioned absolute at top: 0 or left: 0
-    // so we need to transform it to the appropriate location before the animation.
-    // This behavior is because we would otherwise need a double-render in order to
-    // adjust the container position once we had the height the first calendar
-    // (ie first draw all the calendar, then in a second render, use the first calendar's
-    // height to position the container). Variable calendar heights, amirite? <3 Maja
-    this.translateFirstDayPickerForAnimation(translationValue);
+      const newMonthHeight = this.calendarMonthHeights.slice(0, numberOfMonths)
```

Comment:
`const newMonthHeight = Math.max(0, ...this.calendarMonthHeights.slice(0, numberOfMonths));` (the 0 is only necessary because otherwise the max would be `-Infinity` if the array is empty)